test/e2e/cilium_network_policy_test.go

```diff
@@ -0,0 +1,257 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	securityv1 "github.com/openshift/api/security/v1"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/support/azureutil"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TestCiliumConnectivity validates Cilium connectivity using the official Cilium connectivity test suite.
+// This test is specifically for ARO HCP clusters with Cilium as the network provider.
+//
+// The test performs the following steps:
+// 1.Check cilium agent pods ready
+// 2. Creates cilium-test namespace with appropriate labels
+// 3. Deploys Cilium connectivity test pods
+// 4. Waits for all test pods to be ready and running
+// 5. Cleans up test resources.
+func TestCiliumConnectivity(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithCancel(testContext)
+	defer cancel()
+
+	clusterOpts := globalOpts.DefaultClusterOptions(t)
+
+	if globalOpts.Platform != hyperv1.AzurePlatform {
+		t.Skip("Skipping test because it requires Azure platform")
+	}
+
+	if globalOpts.ExternalCNIProvider != "cilium" {
+		t.Skipf("skip cilium connection test if e2e.external-cni-provider is not cilium")
+	}
+
+	e2eutil.NewHypershiftTest(t, ctx, func(t *testing.T, g Gomega, mgtClient crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+		if !azureutil.IsAroHCP() {
+			t.Skip("test only supported on ARO HCP clusters")
+		}
+		guestClient := e2eutil.WaitForGuestClient(t, ctx, mgtClient, hostedCluster)
+
+		testNamespace := "cilium-test"
+
+		// Cleanup function
+		cleanup := func() {
```

Comment:
Instead of putting an anonymous function here, I would create their own cleanup function under `test/e2e/util`. And not just for that, I would create an EnsureXXXXX function for the whole non-cleanup function, so the code would be more readable and the cleanup function could be reused.